### PR TITLE
Add tests for ranged accessors requirement for the entire buffer

### PR DESCRIPTION
**Pull request by vladimirkhashev**
According to [4.7.6.8. Ranged accessors](https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:accessors.ranged)
> A ranged accessor still creates a requisite for the entire underlying buffer, even for the portions not within the range. For example, if one command writes through a ranged accessor to one region of a buffer and a second command reads through a ranged accessor from a non-overlapping region of the same buffer, the second command must still be scheduled after the first because the requisites for the two commands are on the entire buffer, not on the sub-ranges of the ranged accessors.

This test check this requirement.

### CODE REVIEW

File: tests/accessor/accessor_requisite_entire_buffer.cpp [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+//  Provide checks that ranged accessor still creates a requisite for the entire
+//  underlying buffer
+//
+*******************************************************************************/
+
+#include "../common/disabled_for_test_case.h"
+#include "../common/get_cts_object.h"
+#include "catch2/catch_test_macros.hpp"
+
+#include "accessor_common.h"
```
Comment (bader) [inline tests/accessor/accessor_requisite_entire_buffer.cpp:14]: Some of these header files use features not supported by certain implementations. Typical way to fix this issue is:

```c++
// FIXME: re-enable when <FEATURE X> is implemented
#if !SYCL_CTS_COMPILING_WITH_HIPSYCL && !SYCL_CTS_COMPILING_WITH_COMPUTECPP
// includes like 
// #include "accessor_common.h"
#endif
```

**Review by bader — CHANGES_REQUESTED** [review]

File: tests/accessor/accessor_requisite_entire_buffer.cpp (hunk 1 above)
Comment (vladimirkhashev) [reply tests/accessor/accessor_requisite_entire_buffer.cpp]: Thank you, done

File: tests/accessor/accessor_requisite_entire_buffer.cpp [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+#include "catch2/catch_test_macros.hpp"
+
+// FIXME: re-enable when sycl::accessor is implemented
+#if !SYCL_CTS_COMPILING_WITH_HIPSYCL && !SYCL_CTS_COMPILING_WITH_COMPUTECPP
+#include "accessor_common.h"
+#endif
+
+namespace accessor_requisite_entire_buffer {
+
+DISABLED_FOR_TEST_CASE(hipSYCL, ComputeCpp)
```
Comment (bader) [inline tests/accessor/accessor_requisite_entire_buffer.cpp:21]: Could clarify why the test is disabled for these implementations?

**Review by bader — APPROVED** [review]

File: tests/accessor/accessor_requisite_entire_buffer.cpp (hunk 2 above)
Comment (vladimirkhashev) [reply tests/accessor/accessor_requisite_entire_buffer.cpp]: Code that doesn't work with these implementations is included via common file, now this code is disabled in common header file for hipSYCL and function that was disabled for ComputeCpp is enabled. So the test is enabled for all implementations now.

**Review by ProGTX — APPROVED** [review]

**Pull request merged**